src/Package/Impl/Shell/ThemeColorsProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Windows.Media;
+using Microsoft.Common.Wpf.Themes;
+using Microsoft.VisualStudio.PlatformUI;
+
+namespace Microsoft.VisualStudio.R.Package.Shell {
+    [Export(typeof(IThemeColorsProvider))]
+    internal sealed class ThemeColorsProvider : IThemeColorsProvider {
+        public ThemeColorsProvider() {
+            VSColorTheme.ThemeChanged += OnThemeChanged;
+        }
+
+        private void OnThemeChanged(ThemeChangedEventArgs e) {
+            ThemeChanged?.Invoke(this, EventArgs.Empty);
+        }
+
+        public bool IsDarkTheme {
+            get {
+                var defaultBackground = VSColorTheme.GetThemedColor(EnvironmentColors.ToolWindowBackgroundColorKey);
+                return defaultBackground.GetBrightness() < 0.5;
+            }
+        }
+        public Color CodeBackgroundColor {
+            get {
+                return IsDarkTheme ? Color.FromArgb(0xFF, 0x0C, 0x0C, 0x0C) : Color.FromArgb(0xFF, 0xFA, 0xFA, 0xFA);
```

Comment:
Unfortunately it is tricky. Tools | Options only expose editor colorable items, not IDE items. And they are not directly themeable. If one uses standard classifications, they change with themes automatically. IDE colors also change, but they are not exposed to Options (only to Theme Editor). So yes, I will try to figure out if colors can be made themeable, but I am not 100% sure at the moment.